lib/libfrr.c

```diff
@@ -589,6 +618,12 @@ struct thread_master *frr_init(void)
 
 	zprivs_init(di->privs);
 
+	/* Guard to prevent a second instance of this daemon */
+	if (frr_guard_daemon() == FAILURE) {
+		zlog_err("There is already a similar Process running,hence not allowing a second instance.");
```

Comment:
The log message is duplicate since `frr_guard_daemon` already logs the fact that the pid file cannot be locked.

You can remove the return value and make `frr_guard_daemon` return void, and have it directly call `exit(1)`.